assignment/scripts/array-practice.js

```diff
@@ -1,94 +1,113 @@
+console.log('The second animal is: ', animalArray[1]);
 
-// 3. Accessing array items
-console.log('--- 3. Accessing items in an array ---');
+// // 3.b. TODO: Log the last animal in the array using it's array index 
+console.log('The last animal is: ', animalArray[3]);
 
-// Example: Log the first animal from the array using it's array index
-console.log('First animal is', animalArray[0])
 
-// 3.a. TODO: Log the second animal in the array 
+// // 3.c. (STRETCH) TODO: Log the last animal by using the array length, 
+// //      instead of the exact index number of the last item
+console.log(animalArray[animalArray.length-1]);
 
 
-// 3.b. TODO: Log the last animal in the array using it's array index 
+// // 4. Adding & Removing Array Items
+// console.log('--- 4. Adding and removing array items ---');
 
+// // Example: Add an animal to the end using Array.push
+// animalArray.push('penguin');
+// console.log('Added an animal to end,', animalArray);
 
-// 3.c. (STRETCH) TODO: Log the last animal by using the array length, 
-//      instead of the exact index number of the last item
+// // 4.a. TODO: Add a new food at the end of your array & log the array
+favoriteFoods.push('Pecan Pie');
+console.log(favoriteFoods);
 
 
+// // Example: Remove the last animal by using Array.pop
+// let removedAnimal = animalArray.pop();
+// console.log('Removed the last animal', removedAnimal);
+// console.log('The animals are now', animalArray);
 
-// 4. Adding & Removing Array Items
-console.log('--- 4. Adding and removing array items ---');
+// // 4.b. TODO: Remove the food at the end of your array & 
+// //      log both the food removed and the updated array
+removedFoods = favoriteFoods.pop();
+console.log(removedFoods);
 
-// Example: Add an animal to the end using Array.push
-animalArray.push('penguin');
-console.log('Added an animal to end,', animalArray);
+// // Example: Add an animal to the beginning using Array.unshift
+// animalArray.unshift('walrus');
+// console.log(`Added an animal to beginning: ${animalArray}`);
 
-// 4.a. TODO: Add a new food at the end of your array & log the array
+// // 4.c. TODO: Add a food at the beginning of the array & log the array
+favoriteFoods.unshift('Cream Cheese Wontons');
+console.log(favoriteFoods);
+// // Example: Remove the first animal using Array.shift
+// removedAnimal = animalArray.shift();
+// console.log('Removed the first animal', removedAnimal);
+// console.log('The animals are now', animalArray);
 
+// // 4.d TODO: Remove the food at the beginning of your array & 
+// //     log both the food removed and the updated array
+firstRemovedFood = favoriteFoods.shift();
+console.log(firstRemovedFood);
+// // 4.e (STRETCH) TODO: Replace the second food in your array
+// //      with another one of your favorite foods.
+// //      Then log the updated array.
 
+for (let i of favoriteFoods){
+    if(i === favoriteFoods.length-2, favoriteFoods[1]==('Crab Legs'));
```

Comment:
It does not make much sense to put a comma in the middle of if condition, nor to follow the condition with a semicolon. You should write a if block like this:

```js
if (condition) {
  // Code to execute
}
```

If you have multiple conditions, you can combine them with `&&` or `||`.

https://github.com/PrimeAcademy/topaz-syllabus/blob/main/lecture-notes/week-01-html-and-js/02c-conditionals.md

Reply:
Also, fwiw, I think you are overthinking this a bit. This satisfies the prompt:

```js
favoriteFoods[1] = 'Crab Legs';
```